Crash report 3.1: NPE in ImageUtils.createThumbnailFromUri (EditPostPreviewFragment)

```
java.lang.RuntimeException: An error occured while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:299)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:352)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:219)
       at java.util.concurrent.FutureTask.run(FutureTask.java:239)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
       at java.lang.Thread.run(Thread.java:838)
Caused by: java.lang.NullPointerException
       at android.graphics.Bitmap.createBitmap(Bitmap.java:622)
       at org.wordpress.android.util.ImageUtils.createThumbnailFromUri(ImageUtils.java:488)
       at org.wordpress.android.util.ImageUtils.getWPImageSpanThumbnailFromFilePath(ImageUtils.java:350)
       at org.wordpress.android.util.HtmlToSpannedConverter.startImg(WPHtml.java:806)
       at org.wordpress.android.util.HtmlToSpannedConverter.handleStartTag(WPHtml.java:652)
       at org.wordpress.android.util.HtmlToSpannedConverter.startElement(WPHtml.java:993)
       at org.ccil.cowan.tagsoup.Parser.push(Parser.java:794)
       at org.ccil.cowan.tagsoup.Parser.rectify(Parser.java:1061)
       at org.ccil.cowan.tagsoup.Parser.stage(Parser.java:1026)
       at org.ccil.cowan.tagsoup.HTMLScanner.scan(HTMLScanner.java:573)
       at org.ccil.cowan.tagsoup.Parser.parse(Parser.java:449)
       at org.wordpress.android.util.HtmlToSpannedConverter.convert(WPHtml.java:552)
       at org.wordpress.android.util.WPHtml.fromHtml(WPHtml.java:159)
       at org.wordpress.android.util.WPHtml.fromHtml(WPHtml.java:122)
       at org.wordpress.android.ui.posts.EditPostPreviewFragment$LoadPostPreviewTask.doInBackground(EditPostPreviewFragment.java:78)
       at org.wordpress.android.ui.posts.EditPostPreviewFragment$LoadPostPreviewTask.doInBackground(EditPostPreviewFragment.java:62)
       at android.os.AsyncTask$2.call(AsyncTask.java:287)
       at java.util.concurrent.FutureTask.run(FutureTask.java:234)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
       at java.lang.Thread.run(Thread.java:838)
```

--- a/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java
+++ b/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java
@@ -444,7 +444,7 @@ public class ImageUtils {
         optActual.inSampleSize = scale;
 
         // Get the roughly resized bitmap
-        Bitmap bmpResized;
+        final Bitmap bmpResized;
         try {
             bmpResized = BitmapFactory.decodeFile(filePath, optActual);
         } catch (OutOfMemoryError e) {
@@ -452,8 +452,9 @@ public class ImageUtils {
             return null;
         }
 
-        if (bmpResized == null)
+        if (bmpResized == null) {
             return null;
+        }
 
         ByteArrayOutputStream stream = new ByteArrayOutputStream();
 
@@ -490,7 +491,18 @@ public class ImageUtils {
         } catch (OutOfMemoryError e) {
             AppLog.e(AppLog.T.UTILS, "OutOfMemoryError Error in setting image: " + e);
             return null;
+        } catch (NullPointerException e) {
+            // See: https://github.com/wordpress-mobile/WordPress-Android/issues/1844
+            AppLog.e(AppLog.T.UTILS, "Bitmap.createBitmap has thrown a NPE internally. This should never happen: " + e);
+            return null;
         }
+
+        if (bmpRotated == null) {
+            // Fix an issue where bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.
+            // See: https://github.com/wordpress-mobile/WordPress-Android/issues/1848
+            return null;
+        }
+
         bmpRotated.compress(fmt, 100, stream);
         bmpResized.recycle();
         bmpRotated.recycle();

